src/main/java/org/broadinstitute/hellbender/engine/PathURI.java

```diff
@@ -0,0 +1,86 @@
+package org.broadinstitute.hellbender.engine;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.nio.file.Path;
+
+/**
+ * Interface for htsjdk input/output URIs.
+ */
+public interface PathURI {
```

Comment:
I THINK its more of an implementation detail than something that affects the interface itself. I originally included it in the implementation class because I had to keep referring back to it when I first started coding that, and was confused about some of the terminology.